MarkBind/markbind
# [WIP] Make Dropdown menu responsive

**What is the purpose of this pull request?**
• [x] Enhancement to an existing feature
This addresses issue #1314 by enhancing the dropdown to become responsive to window resizing. 

**What is the rationale for this request?**
Currently, the dropdown menu is not responsive to the window resizing. Therefore, when it cannot fit into the current window size, it will be cut in the middle. Refer to issue #1314 for an illustration of the issue. This may affect user experience to some extent. To make it more convenient and usable for users, we should make the dropdown responsive and scrollable upon detection that it cannot fit into the current window size. 

**What changes did you make? (Give an overview)**
I added a ```max-height``` CSS property to ```dropdown-menu``` and made its value dynamic with respect to window resizing. The value is stored in a CSS global variable called ```--max-dropdown-height``` and is computed with ```navbar``` height and current window height. I have also added a small debouncing feature to improve the performance. 

**Is there anything you'd like reviewers to focus on?**
Implementation, code quality, and potential edge cases. 

**Testing instructions:**
1. Create dropdown and populate it with enough content (such that the height of the dropdown menu is more than the smallest size possible for window resizing)
2. Resize the window to be smaller (in height) than the dropdown menu
3. You should find the dropdown menu to shrink in height accordingly
3. Place your cursor within the boundaries of the dropdown menu
4. You should be able to scroll the menu and reach the hidden content

**Proposed commit message: (wrap lines at 72 characters)**
Make Dropdown menu responsive

**To-do**
Create snapshot test cases
